internal/envoy/auth.go

```diff
@@ -16,10 +16,17 @@ package envoy
 import "github.com/envoyproxy/go-control-plane/envoy/api/v2/auth"
 
 // UpstreamTLSContext creates an ALPN h2 enabled TLS Context.
-func UpstreamTLSContext() *auth.UpstreamTlsContext {
+func UpstreamTLSContext(protocol string) *auth.UpstreamTlsContext {
```

Comment:
I'm concerned about adding a parameter here because its a singular parameter, not a list. Ideally these protocols are the _alpn_ protocols, but we're obverloading that behaviour on line 27 to use "tls" as a magic marker to return a blank list of alpn protocols.

Try this instead

```go
func UpstreamTLSContext(protocols ...string) *auth.UpstreamTlsContext {
    return &auth.UpstreamTlsContext{
        CommonTlsContext: &auth.CommonTlsContext{
            AlpnProtocols: protocols,
        },
   },
}
```
Then in see next comment